Havit.Blazor.Documentation/Pages/Components/HxListLayoutDoc/HxListLayout_Demo_NamedViews.razor

```diff
@@ -42,43 +43,46 @@
 </HxListLayout>
 
 @code {
-	[Inject] protected NavigationManager NavigationManager { get; set; }
+    [Inject] protected NavigationManager NavigationManager { get; set; }
 
-	private EditableCultureInfo currentItem;
-	private FilterModelDto filterModel = new FilterModelDto();
-	private HxGrid<EditableCultureInfo> gridComponent;
+    private EditableCultureInfo currentItem;
+    private FilterModelDto filterModel = new FilterModelDto();
+    private HxGrid<EditableCultureInfo> gridComponent;
+    private NamedView<FilterModelDto> selectedNamedView;
 
-	private readonly IEnumerable<NamedView<FilterModelDto>> namedViews = new List<NamedView<FilterModelDto>>()
+    private readonly IEnumerable<NamedView<FilterModelDto>> namedViews = new List<NamedView<FilterModelDto>>()
 		{
 			new NamedView<FilterModelDto>("Minimum = 1", () => new FilterModelDto { MinimumLCID = 1 } ),
 			new NamedView<FilterModelDto>("Minimum = 50", () => new FilterModelDto { MinimumLCID = 50 }),
 			new NamedView<FilterModelDto>("Minimum = 100", () => new FilterModelDto { MinimumLCID = 100 })
 		};
 
-	private List<EditableCultureInfo> localEditableCultureInfos = CultureInfo.GetCultures(CultureTypes.AllCultures).Select(c => new EditableCultureInfo()
+    private List<EditableCultureInfo> localEditableCultureInfos = CultureInfo.GetCultures(CultureTypes.AllCultures).Select(c => new EditableCultureInfo()
 		{
 			LCID = c.LCID,
 			DisplayName = c.DisplayName,
 			Name = c.Name,
 			EnglishName = c.EnglishName
 		}).ToList();
 
-	private Task<GridDataProviderResult<EditableCultureInfo>> LoadDataItems(GridDataProviderRequest<EditableCultureInfo> request)
-	{
-		IEnumerable<EditableCultureInfo> result = localEditableCultureInfos.Where(i => i.LCID >= filterModel.MinimumLCID).ToList();
+    private Task<GridDataProviderResult<EditableCultureInfo>> LoadDataItems(GridDataProviderRequest<EditableCultureInfo> request)
+    {
+        IEnumerable<EditableCultureInfo> result = localEditableCultureInfos.Where(i => i.LCID >= filterModel.MinimumLCID).ToList();
 
-		return Task.FromResult(request.ApplyTo(result));
-	}
+        return Task.FromResult(request.ApplyTo(result));
+    }
 
-	private async Task RefreshDataAsync()
-	{
-		await gridComponent.RefreshDataAsync();
-	}
+    private async Task RefreshDataAsync()
+    {
+        selectedNamedView = namedViews.Where(n => n.Filter.Invoke().MinimumLCID == filterModel.MinimumLCID).FirstOrDefault();
+        await gridComponent.RefreshDataAsync();
+    }
 
-	protected async Task NamedViewSelected(NamedView<FilterModelDto> namedView)
-	{
-		filterModel = namedView.Filter();
-		await gridComponent.RefreshDataAsync();
+    protected async Task NamedViewSelected(NamedView<FilterModelDto> namedView)
+    {
+        filterModel = namedView.Filter();
```

Comment:
Looking at the `Filter()` method which is part of the current implementation, it is obvious that the whole "Named Views feature" needs a wide consolidation (incl. breaking-changes).